kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DataSourceImporter.java

```diff
@@ -258,18 +242,46 @@ private void executeBatchDelete(final Connection connection, final List<DataReco
                 for (int i = 0; i < conditionColumns.size(); i++) {
                     Object oldValue = conditionColumns.get(i).getOldValue();
                     if (null == oldValue) {
-                        log.warn("Old value is null, column={}", conditionColumns.get(i));
+                        log.warn("Record old value is null, record={}", each);
                     }
                     preparedStatement.setObject(i + 1, oldValue);
                 }
                 preparedStatement.addBatch();
             }
-            preparedStatement.executeBatch();
+            int[] executeBatch = preparedStatement.executeBatch();
+            if (executeBatch.length == 0) {
+                log.warn("executeDelete failed, deleteSQL={}, conditionColumns={}", deleteSQL, conditionColumns);
+            }
         } finally {
             batchDeleteStatement = null;
         }
     }
     
+    private void sequentialFlush(final DataSource dataSource, final List<DataRecord> buffer) {
```

Comment:
If there's no unique key, does it run through `sequentialFlush`?

Reply:
I find unique key update, delete event not handle correctly now, i will fix it at next PR.